### PR TITLE
Adding tests for XSS

### PR DESCRIPTION
**Pull request by zmallen**
Work in progress

### CODE REVIEW

File: tests/REQUEST-941-APPLICATION-ATTACK-XSS/941100.yaml [hunk 1]
```diff
@@ -15,12 +15,13 @@
             dest_addr: 127.0.0.1
             method: GET
             port: 80
-            uri: '/demo/xss/xml/vuln.xml.php?input=<script xmlns="http://www.w3.org/1999/xhtml">setTimeout("top.frame2.location=\"javascript:(function () {var x = document.createElement(\\\"script\\\");x.src = \\\"//sdl.me/popup.js?//\\\";document.childNodes\[0\].appendChild(x);}());\"",1000)</script>&//'
+#            uri: '/demo/xss/xml/vuln.xml.php?input=<script xmlns="http://www.w3.org/1999/xhtml">setTimeout("top.frame2.location=\"javascript:(function () {var x = document.createElement(\\\"script\\\");x.src = \\\"//sdl.me/popup.js?//\\\";document.childNodes\[0\].appendChild(x);}());\"",1000)</script>&//'
```
Comment (csanders-git) [inline tests/REQUEST-941-APPLICATION-ATTACK-XSS/941100.yaml]: whats with the hashtag?

File: tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            dest_addr: 127.0.0.1
+            method: GET 
+            port: 80
+            uri: '/foo'
+            headers:
+              User-Agent: ModSecurity CRS 3 Tests
+              Host: localhost
+              Cookie: '%20%20onload%3d%20=941120-3'
+          output:
+            log_contains: '"Matched Data:   onload= found within REQUEST_COOKIES_NAMES:%20%20onload%3d%20:   onload= "'
```
Comment (csanders-git) [inline tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml]: Not sure this is the best way to do this. It doesn't indicate which rule triggered, and I'd expect quite a few would.

Comment (zmallen) [inline tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml]: Yeah i followed this throughout the rules. I am unsure of the best way for us to guarantee we found a log with a specific "id". I was afraid of repeats. Should I just try to use the "id"? At least with this, I definitely know it got through.

Comment (zmallen) [inline tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml]: I'm also thinking that a regex here with such a huge log payload may get tricky. We could possibly take modsec logs, modify them to return lines of json that include an "id" and a "data" key

Comment (csjperon) [inline tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml]: Another option would be to add code to truncate the log before each test. But to do that safely we would need to stop the httpd, truncate the log, start httpd and run the test, check the logs. It also assumes a pretty contained testing environment (nothing else happening while we are running the tests)

**Comment by zmallen** [comment]
status on this?

**Comment by csanders-git** [comment]
Finally got a chance to review both PRs thanks guys!

**Pull request merged**